File: model/DBHelper.vb
Public Class DBHelper
    Public Shared EchoSql As Boolean = False
    
    Public Shared Sub ExecSql(ByVal conn As String, ByVal sql As String)
        If EchoSql Then Console.WriteLine(sql)
        Using cn As New Data.SqlClient.SqlConnection(conn)
            cn.Open()
            Using cm As Data.SqlClient.SqlCommand = cn.CreateCommand()
                cm.CommandText = sql
                cm.ExecuteNonQuery()
            End Using
        End Using
    End Sub

    Public Shared Sub ExecBatchSql(ByVal conn As String, ByVal sql As String)
        Using cn As New Data.SqlClient.SqlConnection(conn)
            cn.Open()
            Using cm As Data.SqlClient.SqlCommand = cn.CreateCommand()
                For Each script As String In sql.Split((vbLf + "GO" + vbCr).Split(","c), System.StringSplitOptions.RemoveEmptyEntries)
                    If script.Trim().Replace(vbCrLf, "").ToUpper() = "GO" Then Continue For
                    If EchoSql Then Console.WriteLine(script)
                    cm.CommandText = script
                    cm.ExecuteNonQuery()
                Next
            End Using
        End Using
    End Sub

    Public Shared Sub DropDb(ByVal conn As String)
        Dim cnBuilder As New SqlClient.SqlConnectionStringBuilder(conn)
        Dim dbName As String = cnBuilder.InitialCatalog
        cnBuilder.InitialCatalog = "master"
        If DbExists(cnBuilder.ToString()) Then
            ExecSql(cnBuilder.ToString(), "ALTER DATABASE " + dbName + " SET SINGLE_USER WITH ROLLBACK IMMEDIATE")
            ExecSql(cnBuilder.ToString(), "drop database " + dbName)

            cnBuilder.InitialCatalog = dbName
            ClearPool(cnBuilder.ToString())
        End If
    End Sub

    Public Shared Sub CreateDb(ByVal conn As String)
        Dim cnBuilder As New SqlClient.SqlConnectionStringBuilder(conn)
        Dim dbName As String = cnBuilder.InitialCatalog
        cnBuilder.InitialCatalog = "master"
        ExecSql(cnBuilder.ToString(), "CREATE DATABASE " + dbName)
    End Sub

    Public Shared Function DbExists(ByVal conn As String) As Boolean
        Dim exists As Boolean
        Dim cnBuilder As New SqlClient.SqlConnectionStringBuilder(conn)
        Dim dbName As String = cnBuilder.InitialCatalog
        cnBuilder.InitialCatalog = "master"

        Using cn As New Data.SqlClient.SqlConnection(cnBuilder.ToString)
            cn.Open()
            Using cm As Data.SqlClient.SqlCommand = cn.CreateCommand()
                cm.CommandText = "select db_id('" + dbName + "')"
                exists = Not cm.ExecuteScalar() Is DBNull.Value
            End Using
        End Using

        Return exists
    End Function

    Public Shared Sub ClearPool(ByVal conn As String)
        Using cn As New Data.SqlClient.SqlConnection(conn)
            Data.SqlClient.SqlConnection.ClearPool(cn)
        End Using
    End Sub

End Class

File: model/DBHelper.cs
using System;
using System.Data.SqlClient;
using System.Diagnostics;
using System.IO;
using Microsoft.SqlServer.Management.Common;
using Microsoft.SqlServer.Management.Smo;
using SchemaZen.Library.Models;

namespace SchemaZen.Library {
	public class DBHelper {
		public static bool EchoSql = false;

		public static void ExecSql(string conn, string sql) {
			if (EchoSql) Console.WriteLine(sql);
			using (var cn = new SqlConnection(conn)) {
				cn.Open();
				using (var cm = cn.CreateCommand()) {
					cm.CommandText = sql;
					cm.ExecuteNonQuery();
				}
			}
		}

		public static void ExecBatchSql(string conn, string sql) {
			using (var cn = new SqlConnection(conn)) {
				cn.Open();
				try
				{
					Server server = new Server(new ServerConnection(cn));
					server.ConnectionContext.ExecuteNonQuery(sql);
				}
				catch (ExecutionFailureException ex)
				{
					throw new SqlBatchException(ex.InnerException as SqlException, 0);
				}
			}
		}

		public static void DropDb(string conn) {
			var cnBuilder = new SqlConnectionStringBuilder(conn);
			var initialCatalog = cnBuilder.InitialCatalog;

		    var dbName = "[" + initialCatalog + "]";

		    if (DbExists(cnBuilder.ToString())) {
				cnBuilder.InitialCatalog = "master";
				ExecSql(cnBuilder.ToString(), "ALTER DATABASE " + dbName + " SET SINGLE_USER WITH ROLLBACK IMMEDIATE");
				ExecSql(cnBuilder.ToString(), "drop database " + dbName);

				cnBuilder.InitialCatalog = initialCatalog;
				ClearPool(cnBuilder.ToString());
			}
		}

		public static void CreateDb(string connection, string databaseFilesPath = null) {
			var cnBuilder = new SqlConnectionStringBuilder(connection);
			var dbName = cnBuilder.InitialCatalog;
			cnBuilder.InitialCatalog = "master";
		    var files = string.Empty;
		    if (databaseFilesPath != null) {
		        Directory.CreateDirectory(databaseFilesPath);
		        files = string.Format(@"ON 
(NAME = {0},
    FILENAME = '{1}\{2}.mdf')
LOG ON
(NAME = {0}_log,
    FILENAME =  '{1}\{2}.ldf')", dbName, databaseFilesPath, dbName + Guid.NewGuid() );
		    }
		    ExecSql(cnBuilder.ToString(), "CREATE DATABASE [" + dbName + "]" + files);
		}

		public static bool DbExists(string conn) {
			var exists = false;
			var cnBuilder = new SqlConnectionStringBuilder(conn);
			var dbName = cnBuilder.InitialCatalog;
			cnBuilder.InitialCatalog = "master";

			using (var cn = new SqlConnection(cnBuilder.ToString())) {
				cn.Open();
				using (var cm = cn.CreateCommand()) {
					cm.CommandText = "select db_id('" + dbName + "')";
					exists = (!ReferenceEquals(cm.ExecuteScalar(), DBNull.Value));
				}
			}

			return exists;
		}

		public static void ClearPool(string conn) {
			using (var cn = new SqlConnection(conn)) {
				SqlConnection.ClearPool(cn);
			}
		}
	}
}
